components/merkledb/src/values.rs

```diff
@@ -261,6 +261,42 @@ impl BinaryValue for [u8; HASH_SIZE] {
     }
 }
 
+/// Concatenates slice of byte buffers prepending size of buffer before each one
+pub fn concat_buffers(buffers: &mut [Vec<u8>]) -> Vec<u8> {
+    use std::convert::TryFrom;
+
+    let mut res_buf = Vec::new();
+    for mut buf in buffers {
+        let size = u32::try_from(buf.len()).unwrap_or_else(|_| panic!("Too large buffer"));
```

Comment:
Is it really a good idea to panic in this case ?

Reply:
I'd assert instead. Panic message isn't much self-describing